src/Credentials/CredentialProvider.php

```diff
@@ -801,6 +811,13 @@ public static function getCredentialsFromSource(
         $credentialSource = !empty($data[$profileName]['credential_source']) ? $data[$profileName]['credential_source'] : null;
         $credentialsPromise = null;
 
+        if ($credentialSource == null && empty($data[$profileName]['role_arn'])) {
```

Comment:
Is this block necessary? If `$credentialSource` is null, it will be caught in the switch block below, and not sure `$data[$profileName]['role_arn']` is ever used by anything past this point?

Reply:
I agree but it was in the SEP: 
"If a role arn is not provided along with credential_source then the SDK must fail / return an error."

As far as placement goes, it has to be in this function since it's public.  Through the default chain, it will never enter this section without a role_arn

Reply:
I did rewrite it to invert the first condition

Reply:
It doesn't really make sense to be checked if it's not actually used by anything in this section. Is there another section for which this check and throw makes sense, where the `role_arn` is actually being used, that might be more what the SEP is referring to?